sway-ir/src/optimize/mem2reg.rs

```diff
@@ -148,19 +151,63 @@ pub fn compute_livein(
     result
 }
 
-/// Promote local values that are accessed via load/store to SSA registers.
-/// We promote only locals of non-copy type, whose every use is in a `get_local`
+// Promote loads of globals constants to SSA registers
+fn promote_globals(context: &mut Context, function: &Function) -> Result<bool, IrError> {
+    let mut replacements = FxHashMap::<Value, Constant>::default();
+    for (_, inst) in function.instruction_iter(context) {
+        if let ValueDatum::Instruction(Instruction {
+            op: InstOp::Load(ptr),
+            ..
+        }) = context.values[inst.0].value
+        {
+            if let ValueDatum::Instruction(Instruction {
+                op: InstOp::GetGlobal(global_var),
+                ..
+            }) = context.values[ptr.0].value
+            {
+                if !global_var.is_mutable(context)
+                    && is_promotable_type(context, global_var.get_inner_type(context))
+                {
+                    let constant = *global_var
+                        .get_initializer(context)
+                        .expect("We're dealing with an uninitialized value");
```

Comment:
The proposal is to document here why the `GlobalVar` must be initialized here, because it doesn't need to be.
```suggestion
                        .expect("`global_var` is not mutable so it must be initialized");
```